src/main/java/tdl/client/runner/RoundManagement.java

```diff
@@ -14,15 +14,15 @@
     private static final Path CHALLENGES_FOLDER = Paths.get("challenges");
     private static final Path LAST_FETCHED_ROUND_PATH = CHALLENGES_FOLDER.resolve("XR.txt");
 
-    static void saveDescription(String rawDescription, IConsoleOut consoleOut) {
+    static void saveDescription(boolean recordingSystemOn, String rawDescription, IConsoleOut consoleOut) {
         // DEBT - the first line of the response is the ID for the round, the rest of the responseMessage is the description
         int newlineIndex = rawDescription.indexOf('\n');
         if (newlineIndex <= 0) return;
 
         String roundId = rawDescription.substring(0, newlineIndex);
         String lastFetchedRound = getLastFetchedRound();
         if (!roundId.equals(lastFetchedRound)) {
-            RecordingSystem.notifyEvent(roundId, getNewRoundDescription.getShortName(), consoleOut);
+            RecordingSystem.notifyEvent(recordingSystemOn, roundId, getNewRoundDescription.getShortName());
```

Comment:
The RoundManagement should now know about the RecordingSystem, we could introduce an interface RoundChangesListener, with a method `onNewRound(roundId, shortName)`

Reply:
Thinking about it, I think RoundManagement used to take a callback, and then I went through a point where I removed all callbacks....so yes, this is a good idea.